### PR TITLE
Adds time format constant and some changees at from_json method

### PR DESCRIPTION
**Pull request by GabrielMalakias**

### CODE REVIEW

**Comment by tfausak** [comment]
Thanks for opening this PR! You can ignore the build failure; Rubinius is having trouble. 

Can you explain your motivation for making these changes?

**Comment by GabrielMalakias** [comment]
Ok, well I saw this unfrozen string then I checked if Ruby already has a predefined constant to use it but hasn't. I defined the time_format constant to avoid string creation when executes from_json. After that I created an object to avoid access ['error'] in hash two times. 

I decided do this pull request because I'm using it then I see this to do improvements. It hasn't a big thing but is an improvement, I think.

File: spec/stoplight/default_spec.rb [hunk 1]
```diff
@@ -77,4 +77,14 @@
...
     end
   end
+
+  describe '::TIME_FORMAT' do
+    it 'is a string' do
+      expect(Stoplight::Default::TIME_FORMAT).to be_a(String)
+    end
+
+    it 'is frozen' do
+      expect(Stoplight::Default::NOTIFIERS).to be_frozen
```
Comment (tfausak) [inline spec/stoplight/default_spec.rb]: This is checking the wrong value.

File: lib/stoplight/default.rb [hunk 2]
```diff
@@ -23,5 +23,7 @@ module Default
     ].freeze
 
     THRESHOLD = 3
+
+    TIME_FORMAT = '%Y-%m-%dT%H:%M:%S.%N%:z'.freeze
```
Comment (tfausak) [inline lib/stoplight/default.rb]: The `Default` module is meant for things that can be changed by the user. Would you mind moving this constant into the `Failure` module?

File: spec/stoplight/default_spec.rb (hunk 1 above)
Comment (GabrielMalakias) [inline spec/stoplight/default_spec.rb]: Sorry.. :)

File: lib/stoplight/default.rb (hunk 2 above)
Comment (GabrielMalakias) [inline lib/stoplight/default.rb]: Ok, I'm going to move it. Thanks!

**Pull request merged**